src/components/Form/CrossPodWriteForm.jsx

```diff
@@ -20,69 +19,50 @@ import FormSection from './FormSection';
  *
  * @memberof Forms
  * @name CrossPodWriteForm
+ * @returns {React.ReactElement} The form
  */
-
 const CrossPodWriteForm = () => {
   const { session } = useSession();
   const { state, dispatch } = useStatusNotification();
-  const { selectedUser } = useContext(SelectUserContext);
-  const [username, setUsername] = useState();
   const [docType, setDocType] = useState('');
+  const [expiryDate, setExpiryDate] = useState('');
+  const [description, setDescription] = useState('');
+  const [file, setFile] = useState('');
+  const { podUrl, person } = useContext(SelectUserContext);
 
   const handleDocType = (event) => {
     setDocType(event.target.value);
   };
 
-  // Initalized state for file upload
-  const handleFileChange = (event) => {
-    if (event.target.files.length === 1) {
-      dispatch({ type: 'SET_FILE', payload: event.target.files[0] });
-    } else {
-      dispatch({ type: 'CLEAR_FILE' });
-    }
-  };
-
-  // Custom useField hook for handling form inputs
-  const { clearValue: clearDescription, _type, ...description } = useField('textarea');
-
-  const clearInputFields = (event) => {
-    event.target.reset();
-    clearDescription();
+  const clearInputFields = () => {
+    setDocType('');
+    setExpiryDate('');
+    setDescription('');
+    setFile(null);
     dispatch({ type: 'CLEAR_FILE' });
+    dispatch({ type: 'CLEAR_VERIFY_FILE' });
     dispatch({ type: 'CLEAR_PROCESSING' });
   };
 
-  const handleFormSubmit = makeHandleFormSubmission(
-    INTERACTION_TYPES.CROSS,
-    state,
-    dispatch,
-    session,
-    clearInputFields
-  );
-
   // Event handler for form/document submission to Pod
-  const handleCrossPodUpload = async (event) => {
-    event.preventDefault();
-    dispatch({ type: 'SET_PROCESSING' });
-    const podUsername = event.target.crossPodUpload.value || selectedUser.username;
-
-    if (!podUsername) {
-      runNotification('Search failed. Reason: Username not provided.', 5, state, dispatch);
-      setTimeout(() => {
-        dispatch({ type: 'CLEAR_PROCESSING' });
-      }, 3000);
-      return;
-    }
+  const handleFormSubmission = async (e) => {
+    e.preventDefault();
+    const fileDesc = {
+      name: file.name.split('.')[0],
+      type: docType,
+      date: expiryDate,
+      description
+    };
+    runNotification(`Uploading "${file.name}" to Solid...`, 3, state, dispatch);
 
-    if (!docType) {
-      runNotification('Search failed. Reason: No document type selected.', 5, state, dispatch);
-      setTimeout(() => {
-        dispatch({ type: 'CLEAR_PROCESSING' });
-      }, 3000);
-      return;
+    try {
+      await createDocument(file, fileDesc, session, `${podUrl}PASS/`);
```

Comment:
Function seems to bug out when trying to upload files.

![Screen Shot 2023-06-14 at 14 35 23 (2)](https://github.com/codeforpdx/PASS/assets/14917816/d8a815db-0f4e-407b-bbba-26180d9f3c1c)

It looks like it's trying to build a new sub container within Bank Statement, but couldn't (see Pod after trying to upload). So it couldn't really place the file in it.

![Screen Shot 2023-06-14 at 14 32 52 (2)](https://github.com/codeforpdx/PASS/assets/14917816/1d8ffafa-7ce6-45c7-b422-76caa9f7990f)

Reply:
That's an authentication error. That will happen when you don't have authorization to write to a container. Does your user have access to the pod it's trying to write to?

We may need to work on case manager permissions in another PR, but an upload failing when a user doesn't have permission to access the pod is expected behavior.

Reply:
Oh, it looks like I forgot to pass the session fetch to one of the network calls. Fixed it. This should work now.